### DISCUSSION TITLE
Caching files to local disk

### DISCUSSION THREAD

**Discussion by valentin-iris** [RaRe-Technologies/smart_open#676] [Q&A] [answered]
Hi folks! I was wondering if there is the possibility to cache files on local disks when getting them from e.g. S3. Imagine I have a test or sth that I run very often and a medium to large file on S3 that is needed for it. Is it possible for smart_open to cache that file on local disk (such that I don't have to redownload it every time) and only redownloads it if the file changes on S3 (e.g. via checking file hashes)?
Cheers, Valentin

**Comment by piskvorky** [comment, marked as answer]
Of course! But that's something you'd do on top of `smart_open` (and rather easily, I think). Definitely not as a part of core `smart_open`.

**Reply by valentin-iris** [reply]
Thanks for your answer @piskvorky! I see. Would you mind giving me some pointers as to how to go about this? Cheers
